### PR TITLE
Dev chile visualisation

### PR DESCRIPTION
**Pull request by ao559x**

### CODE REVIEW

**Review by msff2 — COMMENTED** [review]
Address the current comments now and I will have more comments after running the stack.

File: Deploy/stacks/Chile/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+
+The folders to populate will be found in TheWorldAvatar/Deploy/stacks/dynamic/
+
+Details:
+
+* TheWorldAvatar/Deploy/stacks/dynamic/stack-manager/inputs/secrets/
+
+  
+File names:
```
[the remark this answers is not in the document]
Comment (ao559x) [reply Deploy/stacks/Chile/README.md:42]: Transfer the following files to the secrets folder and replace the content accordingly:

File: Deploy/stacks/Chile/stack-manager/inputs/config/services/geoserver.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
+    "type": "geoserver",
+    "ServiceSpec": {
+        "Name": "geoserver",
+        "TaskTemplate": {
+            "ContainerSpec": {
+                "Image": "docker.osgeo.org/geoserver:2.23.1",
+                "Env": [
+                    "GEOSERVER_CSRF_WHITELIST=146.190.86.59",
```
Comment (msff2) [inline Deploy/stacks/Chile/stack-manager/inputs/config/services/geoserver.json:9]: Replace 146.190.86.59 with ${IP_ADDRESS}

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/A1windpotential.json [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    "name": "Chile",
+    "groups": [
+                {
+            "name": "Wind farm suitability Cyan:good, Magenta:bad, Yellow:allowed",
+            "sources": [
+                {
+                    "id": "calc1atlas_capacity_iec1-1",
+                    "type": "raster",
+                    "tiles": [
+                        "http://146.190.86.59:3852/geoserver/ows?service=WMS&version=1.1.0&request=GetMap&layers=the_world_avatar:calc1atlas_capacity_iec1&bbox={bbox-epsg-3857}&width=256&height=256&srs=EPSG:3857&transparent=true&format=image/png"
```
Comment (msff2) [inline Deploy/stacks/Chile/stack-manager/inputs/data/webspace/A1windpotential.json:11]: Replace 146.190.86.59:3852 with ${IP_ADDRESS}:${PORT} here in all files where it appears.

File: Deploy/stacks/Chile/README.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+* mapbox_api_key        (get a token at the mapbox website)
+* mapbox_username       (get a username at the mapbox website)
+
+
+### Step 3. When the folders are populated, use Visual Studio Code to access the folder where TheWorldAvatar is cloned.
+
+* Go to the folder:
+  
+        TheWorldAvatar/Deploy/stacks/dynamic/stack-manager/
+
```
Comment (msff2) [inline Deploy/stacks/Chile/README.md:57]: Put an instruction at this level to set two environment variables, one for IP_ADDRESS and the other for PORT. Instruct users to set these variables using the command below. Put also an example for each variable.

export IP_ADDRESS=<IP_ADDRESS>
export PORT=<PORT>

File: Deploy/stacks/Chile/stack-manager/inputs/config/services/geoserver.json (hunk 2 above)
Comment (ao559x) [reply Deploy/stacks/Chile/stack-manager/inputs/config/services/geoserver.json]: done

File: Deploy/stacks/Chile/README.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+* geoserver_password   (any password for admin)
+* postgis_password     (any password for postgres, server <STACK_NAME>-postgis:5432 for example chileViv4-postgis:5432, system PostgreSQL)
+* mapbox_api_key        (get a token at the mapbox website)
+* mapbox_username       (get a username at the mapbox website)
+
+
+
+
+### Step 3. When the folders are populated, use Visual Studio Code to access the folder where TheWorldAvatar is cloned.
+
```
[the remark this answers is not in the document]
Comment (msff2) [reply Deploy/stacks/Chile/README.md:55]: Also instruct Linux users to go to the folder TheWorldAvatar/Deploy/stacks/dynamic/stack-manager/inputs/data/webspace/ in the VS Code terminal and run the following command to replace <IP_ADDRESS>:<PORT> in all those files where it appears with the actual IP address, e.g., 127.0.0.1 and port, e.g., 3838.

find . -type f -exec sed -i 's/<IP_ADDRESS>:<PORT>/127\.0\.0\.1:3838/g' {} +

[the remark this answers is not in the document]
Comment (ao559x) [reply Deploy/stacks/Chile/README.md:55]: I am going to use:
Please execute:
sed -i 's\<IP_ADDRESS>:<PORT_NUMBER>\<IP_ADDRESS>:<PORT_NUMBER>\g' *.json  
sed -i 's\<IP_ADDRESS>:<PORT_NUMBER>\<IP_ADDRESS>:<PORT_NUMBER>\g' *.html  

For example:
sed -i 's\<IP_ADDRESS>:<PORT_NUMBER>\127.0.0.1:3838\g' *.json  
sed -i 's\<IP_ADDRESS>:<PORT_NUMBER>\127.0.0.1:3838\g' *.html

File: Deploy/stacks/Chile/README.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+With the configuration files found in GitHub at:
+
+* TheWorldAvatar/Deploy/stacks/Chile/ (in the repository at GitHub)
+
+
+The folders to populate will be found in TheWorldAvatar/Deploy/stacks/dynamic/
+
+Go to the following folder:
+
+* TheWorldAvatar/Deploy/stacks/dynamic/stack-manager/inputs/secrets/    (this is were you upload your passwords for geoserver, postgis and mapbox)
```
Comment (msff2) [inline Deploy/stacks/Chile/README.md:41]: Replace "(this is were you upload your passwords for geoserver, postgis and mapbox)" with "(this is where you provide your passwords for GeoServer and PostGIS, and API Key for Mapbox)"

File: Deploy/stacks/Chile/README.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+### Step 7. If you need to delete the stack before that:
+
+        ./stack.sh rm <STACK_NAME> -v
+
+For example:
+
+        ./stack.sh rm chileViv4 -v
+
+### Step 8. In the code folder you can find useful queries to be runned in the adminer.
```
Comment (msff2) [inline Deploy/stacks/Chile/README.md:120]: Remove this step.

File: Deploy/stacks/Chile/README.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+        ./stack.sh rm <STACK_NAME> -v
+
+For example:
+
+        ./stack.sh rm chileViv4 -v
+
+### Step 8. In the code folder you can find useful queries to be runned in the adminer. 
+
+### Step 9. To activate all the options of Geoserver.
```
Comment (msff2) [inline Deploy/stacks/Chile/README.md:122]: Remove Step 9

File: Deploy/stacks/Chile/README.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+For example:
+
+        cd ..
+        cd stack-data-uploader 
+        ./stack.sh start chileViv4
+  
+(This should upload all the information. It might take 10-30 minutes. Be patient.)
+
+### Step 6. If everything goes well, you should be able to see the visualisation at:
+* http://<IP_ADDRESS>:<PORT_NUMBER>/visualisation/welcome.html   (forwards to website)
```
Comment (msff2) [inline Deploy/stacks/Chile/README.md:101]: This URL does not work, so please fix the issue.

File: Deploy/stacks/Chile/README.md (hunk 7 above)
Comment (ao559x) [reply Deploy/stacks/Chile/README.md:120]: if I delete this, then were do we say how to use the queries?

File: Deploy/stacks/Chile/README.md (hunk 6 above)
Comment (ao559x) [reply Deploy/stacks/Chile/README.md:41]: done

File: Deploy/stacks/Chile/README.md (hunk 8 above)
Comment (ao559x) [reply Deploy/stacks/Chile/README.md:122]: What if we call them additional features?

File: Deploy/stacks/Chile/README.md (hunk 7 above)
Comment (msff2) [reply Deploy/stacks/Chile/README.md]: Andrea has moved this to the Additional Features section, which is fine.

**Comment by ao559x** [comment]
Hello. Maybe before merging Jethro @jwja2  may want to take a look to the text in the following file:
Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html

Output available on https://theworldavatar.io/demos/chile/
I changed visualization by visualisation.

File: Deploy/stacks/Chile/README.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+### Step 4. When the folders are populated, use Visual Studio Code to access the folder where TheWorldAvatar is cloned.
+
+
+* Go to the folder:
+  
+        TheWorldAvatar/Deploy/stacks/dynamic/stack-manager/
+
+* Run the following command with the actual stack name and port number to spin up a stack:
+
+         ./stack.sh start <STACK_NAME> <PORT_NUMBER>
```
Comment (msff2) [inline Deploy/stacks/Chile/README.md:75]: As you have fixed the stack name to be chileViv4 by providing a file called chileViv4.json under the path of TheWorldAvatar/Deploy/stacks/dynamic/stack-manager/inputs/config, replace ./stack.sh start <STACK_NAME> <PORT_NUMBER> with ./stack.sh start chileViv4 <PORT_NUMBER>. Also, note that changing the name of the stack requires changing the name of this JSON file.

File: Deploy/stacks/Chile/stack-data-uploader/inputs/README.md [hunk 11]
```diff
@@ -0,0 +1,4 @@
+UPDATED ON 29 February 2024.
```
[the remark this answers is not in the document]
Comment (ao559x) [reply Deploy/stacks/Chile/stack-data-uploader/inputs/README.md]: Line 1 has been deleted

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/data.json [hunk 12]
```diff
@@ -0,0 +1,695 @@
...
+    "name": "NHS Sites",
+    "groups": [
+        {
+            "name": "England",
+            "sources": [
+                {
+                    "id": "gp-source",
+                    "type": "vector",
+                    "tiles": [
+                        "http://146.190.86.59:3838/geoserver/ows?service=WMS&version=1.1.0&request=GetMap&layers=the_world_avatar:nhs_gp_practices&bbox={bbox-epsg-3857}&width=256&height=256&srs=EPSG:3857&format=application/vnd.mapbox-vector-tile"
```
[the remark this answers is not in the document]
Comment (ao559x) [reply Deploy/stacks/Chile/stack-manager/inputs/data/webspace/data.json]: changed to <IP_ADDRESS>:<PORT_NUMBER>

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/READMEhelp.md
[the remark this answers is not in the document]
Comment (ao559x) [reply Deploy/stacks/Chile/stack-manager/inputs/data/webspace/READMEhelp.md:1]: Deleted and contents added to main README

**Review by sm453 — APPROVED** [review]
Fine from my side.

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html [hunk 13]
```diff
@@ -0,0 +1,389 @@
...
+following document </span><span style='font-size:16.0pt;line-height:107%'><a
+href="http://theworldavatar.io/demos/chile/guide.pdf"><span
+style='font-family:"Verdana",sans-serif;color:#0B4452'>VISUALISING GUIDE</span></a></span><span
+style='font-size:16.0pt;line-height:107%;font-family:"Verdana",sans-serif'>.
+The following <i>User Targeted Visualisations</i> have been developed:</span></p>
+
+<p class=MsoNormal><span style='font-size:16.0pt;line-height:107%;font-family:
+"Verdana",sans-serif'>&nbsp;</span></p>
+
+<p class=MsoListParagraph style='text-indent:-18.0pt'><b><span
```
Comment (jwja2) [inline Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html:171]: The names of the visualisations need to match the names that we used for them in the Supplementary Information.

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html [hunk 14]
```diff
@@ -0,0 +1,389 @@
...
+"Verdana",sans-serif'>&nbsp;</span></p>
+
+<p class=MsoNormal><span style='font-size:16.0pt;line-height:107%;font-family:
+"Verdana",sans-serif'>&nbsp;</span></p>
+
+<p class=MsoNormal><span style='font-size:16.0pt;line-height:107%;font-family:
+"Verdana",sans-serif'>In this visualisation, we have addressed Chile's
+energy grid as a use case, given the role this country might have in the coming
+years as a renewable energy producer. A guided tour is available in the
+following document </span><span style='font-size:16.0pt;line-height:107%'><a
```
Comment (jwja2) [inline Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html:162]: Should say "in the following visualisation guide"

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html [hunk 15]
```diff
@@ -0,0 +1,389 @@
...
+<p class=MsoNormal>&nbsp;</p>
+
+<p class=MsoNormal><span style='font-size:24.0pt;line-height:107%'>&nbsp;</span></p>
+
+<p class=MsoNormal><span style='font-size:16.0pt;line-height:107%;font-family:
+"Verdana",sans-serif'>&nbsp;</span></p>
+
+<p class=MsoNormal><span style='font-size:16.0pt;line-height:107%;font-family:
+"Verdana",sans-serif'>Sustainability is one of the most important 21<sup>th</sup>
+century challenges to be addressed. However, given the problem is
```
Comment (jwja2) [inline Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html:136]: This should be "21st Century". (So change the suffix on the number and capitalise century).

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html (hunk 14 above)
Comment (ao559x) [reply Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html]: done

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html (hunk 15 above)
Comment (ao559x) [reply Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html]: done

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html (hunk 13 above)
Comment (ao559x) [reply Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html:170]: done

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html (hunk 15 above)
Comment (jwja2) [reply Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html]: Confirmed.

File: Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html (hunk 14 above)
Comment (jwja2) [reply Deploy/stacks/Chile/stack-manager/inputs/data/webspace/index.html]: the words "visualisation guide" should be the link. At the minute it reads, ..."the following visualisation guide VISUALISING GUIDE".

**Pull request merged**